app/src/main/kotlin/com/mapbox/vision/teaser/SafetyFragment.kt

```diff
@@ -0,0 +1,238 @@
+package com.mapbox.vision.teaser
+
+import android.animation.Animator
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.animation.AccelerateDecelerateInterpolator
+import androidx.fragment.app.Fragment
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants
+import com.mapbox.services.android.navigation.v5.utils.DistanceFormatter
+import com.mapbox.services.android.navigation.v5.utils.LocaleUtils
+import com.mapbox.vision.mobile.core.models.Country
+import com.mapbox.vision.mobile.core.models.detection.DetectionClass
+import com.mapbox.vision.safety.VisionSafetyManager
+import com.mapbox.vision.safety.core.VisionSafetyListener
+import com.mapbox.vision.safety.core.models.CollisionDangerLevel
+import com.mapbox.vision.safety.core.models.CollisionObject
+import com.mapbox.vision.safety.core.models.RoadRestrictions
+import com.mapbox.vision.teaser.models.UiSign
+import com.mapbox.vision.teaser.utils.SoundsPlayer
+import com.mapbox.vision.teaser.utils.classification.SignResources
+import com.mapbox.vision.teaser.utils.requireMainActivity
+import com.mapbox.vision.teaser.utils.requireVisionManager
+import com.mapbox.vision.teaser.utils.runOnUiThread
+import com.mapbox.vision.teaser.view.hide
+import com.mapbox.vision.teaser.view.show
+import kotlinx.android.synthetic.main.fragment_safety.*
+
+class SafetyFragment : Fragment() {
+
+    companion object {
+        val TAG: String = SafetyFragment::class.java.simpleName
+        private const val CALIBRATION_READY_VALUE = 1f
+        fun newInstance() = SafetyFragment()
+    }
+
+    private lateinit var soundsPlayer: SoundsPlayer
+    var calibrationProgress = 0F
```

Comment:
I can't find where your update this field. Only read it

Reply:
https://github.com/mapbox/mapbox-vision-android-teaser/blob/ybk_extract_safety_fragment/app/src/main/kotlin/com/mapbox/vision/teaser/MainActivity.kt#L120

Reply:
maybe hide this setter behind the interface
@yunikkk @kiryldz wdyt?

Reply:
I'd say it's a matter of taste.
Personally I do not like those non-private `var`-s getting set from somewhere - I'd go with interface or even just method.
BTW progress could be set and read from different threads - you might want to take it into account.
@yuryybk - just as another idea - could we make whole `SafetyFragment` implement `VisionSafetyListener`?

Reply:
There is no reason to make SafetyFragment implementing VisionSafetyListener.
As we discussed in the voice, I'll extract other Vision Features to Fragments and will see how it looks. Probably will add some interfaces later